src/bp/ui-shared/src/FormFields/SuperInput/index.tsx

```diff
@@ -51,20 +49,21 @@ export default ({ canAddElements, events, variables, setCanOutsideClickClose, on
     },
     ['dropdown:show']: e => {
       setCanOutsideClickClose?.(false)
-    }
-  }
+    },
+    ['edit:start']: e => {
+      const prefix = e.detail.data.prefix
 
-  const sanitizeTagSearch = tag => {
-    const currentContent = tagifyRef.current?.DOM?.input?.innerHTML
-    const currentTag = currentContent.substring(currentContent.length - tag.length)
-    const tagIndex = currentContent.lastIndexOf(currentTag)
-    const newContent =
-      currentContent.substring(0, tagIndex) +
-      sanitizeName(currentTag) +
-      currentContent.substring(tagIndex + currentTag.length)
-    // @ts-ignore
-    tagifyRef.current?.DOM?.input?.innerHTML = newContent
-    moveCarretToEndOfString()
+      if (prefix) {
+        if (prefix == '$') {
+          tagifyRef.current.settings.whitelist = variables?.map(({ name }) => name) || []
+        }
+
+        if (prefix == '{{') {
+          // TODO refactor to use the schema format properly and allow to breakdown into an object type search
+          tagifyRef.current.settings.whitelist = events?.map(event => event.name) || []
+        }
+      }
+    }
```

Comment:
Please use strict equality ( === ). Also, your first prefix if() is redundant, an undefined prefix would not match either conditions